packages/firestore/test/unit/specs/recovery_spec.test.ts

```diff
@@ -114,6 +136,31 @@ describeSpec('Persistence Recovery', ['no-ios', 'no-android'], () => {
     }
   );
 
+  specTest(
+    'Ignores intermittent lease refresh failures (with recovery)',
+    ['multi-client'],
+    () => {
+      return client(0)
+        .expectPrimaryState(true)
+        .client(1)
+        .expectPrimaryState(false)
+        .client(0)
+        .failDatabaseTransactions('updateClientMetadataAndTryBecomePrimary')
+        .runTimer(TimerId.ClientMetadataRefresh)
+        .client(1)
+        .failDatabaseTransactions('updateClientMetadataAndTryBecomePrimary')
```

Comment:
I'm sort of surprised we need to specify this on a per-client basis. My expectation would be that if the database is failing it would fail for all clients, no?

Do you find that you're writing tests where different clients see different failures? I wonder if we could make that the exception, rather than the rule?

Reply:
There are already a couple multi-tab tests in this file that use different failure modes per client, which also matches my expected behavior. A foreground tab is able to access IndexedDB, while a background tab lost filesystem access.

FWIW, I am not planning on adding further multi-tab tests for now.